pygfx/objects/_base.py

```diff
@@ -173,12 +168,43 @@ def __init__(
         self.cast_shadow = False
         self.receive_shadow = False
 
+    @callback
+    def _update_uniform_buffers(self, transform: AffineBase):
+        self.uniform_buffer.data["world_transform"] = transform.matrix.T
+        self.uniform_buffer.data["world_transform_inv"] = transform.inverse_matrix.T
+        self.uniform_buffer.update_range()
+
     def __repr__(self):
         return f"<pygfx.{self.__class__.__name__} at {hex(id(self))}>"
 
     def __del__(self):
         id_provider.release_id(self, self.id)
 
+    @property
+    def up(self):
+        """
+        Relic of old WorldObjects that aliases with the new ``transform.up``
+        direction. Prefer (minus) `-obj.world.gravity.` instead
+
+        """
```

Comment:
The problem with using `up` is that it is aliased by `transform.up` which represents the "up" direction of the source represented in the target frame (`(0, 1, 0)` transformed by `self`). 

We need a word for it that is sufficiently different to avoid confusion and to not define the same property twice. I think it is okay to break with tradition here.

Reply:
The downside is that it implies we are offering some sort of physics simulation. Gravity is usually the variable that affects falling speed and the like.

Reply:
@Korijn I'm happy to call it something other than `gravity`. Just not `up`, because the name is taken and I don't think we want to rename the direction `up` to something else 😄

Reply:
IIUC `gravity` is used as a reference, and `up` is computed/synchronized in relation to the transform. So I would suggest `up_reference` like you've done before.

Reply:
I Googled around a bit ...

In Unity, they also use `up` for the same things as we do (in this PR). You can change the direction of gravity, which seems to be a vector for which the length signifies the strength, and is in the physics settings.

In blender they have a setting to change what is up, using a setting called gravity: https://blender.stackexchange.com/questions/471/is-it-possible-to-make-blender-a-y-up-world but if you google "blender gravity" you'll find that it's also a scalar setting to set the gravity strength.

In short, and in hindsight, I don't think the term "up" is already familiar except for ppl that know ThreeJS. I also see how "gravity" may sounds a bit too physics-like. The `up_reference` could work. Some other ideas: "top", "upwards" ...

Reply:
I doubt `gravity` will be the first property users will examine when trying to accomplish what they _can_ accomplish with the property. This property should be named such that users will naturally reach for it when they want to maintain an upright orientation of objects, cameras and lights, when making them face certain directions/look at things.

Can you give a crystal clear explanation of the difference in _purpose_ of the `up` and `gravity` properties in this PR? I feel like whatever names we choose, they should reflect the intended purpose.

For `up` it's very clear to me that it's the current upward direction of an object in its parent space. We shouldn't change this.

For `gravity`, it's less clear. It's not down because then it would be called `down`. :) If I understand correctly it's meant to be used precisely as a reference for `up`. As if we are making a sort of promise to the user that we shall strive to keep `up` aligned with `-gravity` as much as we possibly can. For that reason I think `up_reference` is pretty good!

But as an alternative I can think of things like `up_desired`, `up_constant`, `world_up`, `reference_up`, `target_up`, `lookat_up`.

I wrote so much here in an effort to explain myself and my understanding, such that you can point out where I am missing the point. :)

Reply:
> Can you give a crystal clear explanation of the difference in purpose of the `up` and `gravity` properties [...]

Not sure if you asked Sebastian or me, but I can answer from my perspective :)

The `ob.world.gravity` was formerly `ob.up`, and meant the same as it did in ThreeJS. The ThreeJS docs are a bit minimal on it: *This is used by the `lookAt` method, for example, to determine the orientation of the result.*

I think your explanation covers it pretty well. More precisely, I think it coverts three points:
* In calling `look_at` the object's up-side is kept in the [general direction](https://www.youtube.com/watch?v=FWBUl7oT9sA) of `up_reference`.
* The camera's `show_x` methods work similarly: position the camera in this way, keeping the camera's up-side in *this* general direction. They all call `look_at` in the end to achieve this.
* Some controllers use the camera's `up_reference` to define certain rotations, e.g. the orbit camera.

> `reference_up`

+1 for this one as it is less ambiguous than `up_reference`. We could also do `whats_up` which is actually pretty accurate, though probably a bit too playful 😄

Reply:
In that case, I propose we go with `up_reference`.

Reply:
Hm. I just updated the docstring for `look_at` and settled for:

```
        This sets the object's rotation such that its ``forward`` direction
        points towards ``target`` (given in world space). This rotation takes
        gravity into account, i.e., the rotation is chosen in such a way that a
        camera looking ``forward`` follows the rotation of a human head looking
        around without tilting the head sideways.
```

Considering this, what do you think of `body_direction` or `main_body_direction`?

Reply:
Hm, `upright` could work, but it kind of looses its meaning once `forward` leaves the XZ-plane (why is `upright` no longer "upright"?).

I like `neutral_tilt` but would expect it to be an angle ("which tilt to we consider to be neutral"). It's also a bit odd to explain why the `neutral_tilt` vector isn't normal to the viewing axis most of the time, since `tilt` would be measured around the viewing axis (aka. forward).

My favorite is still `body_direction`, but I guess something like `tilt_reference` could do the trick as well. Like "We measure tilt around the viewing axis (forward) with the `tilt_reference` (projected into the viewing plane) marking zero tilt." If we want to lean more onto established nomenclature, we could also call it `roll_reference` since we are talking about `roll` in the pitch, roll, yaw formulation.